src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl

```diff
@@ -0,0 +1,403 @@
+    A = compute_a_coeffs(num_stages, stage_scaling_factors, monomial_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    return a_matrix, c
+end
+
+# Compute the Butcher tableau for a paired explicit Runge-Kutta method order 2
+# using provided monomial coefficients file
+function compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                   base_path_monomial_coeffs::AbstractString,
+                                                   bS, c_end)
+
+    # c Vector form Butcher Tableau (defines timestep per stage)
+    c = zeros(num_stages)
+    for k in 2:num_stages
+        c[k] = c_end * (k - 1) / (num_stages - 1)
+    end
+    stage_scaling_factors = bS * reverse(c[2:(end - 1)])
+
+    # - 2 Since first entry of A is always zero (explicit method) and second is given by c_2 (consistency)
+    coeffs_max = num_stages - 2
+
+    a_matrix = zeros(coeffs_max, 2)
+    a_matrix[:, 1] = c[3:end]
+
+    path_monomial_coeffs = joinpath(base_path_monomial_coeffs,
+                                    "gamma_" * string(num_stages) * ".txt")
+
+    @assert isfile(path_monomial_coeffs) "Couldn't find file"
+    monomial_coeffs = readdlm(path_monomial_coeffs, Float64)
+    num_monomial_coeffs = size(monomial_coeffs, 1)
+
+    @assert num_monomial_coeffs == coeffs_max
+    A = compute_a_coeffs(num_stages, stage_scaling_factors, monomial_coeffs)
+
+    a_matrix[:, 1] -= A
+    a_matrix[:, 2] = A
+
+    return a_matrix, c
+end
+
+"""
+    PairedExplicitRK2()
+
+The following structures and methods provide a minimal implementation of
+the second-order paired explicit Runge-Kutta (PERK) method
+optimized for a certain simulation setup (PDE, IC & BC, Riemann Solver, DG Solver).
+
+- Brian Vermeire (2019).
+  Paired explicit Runge-Kutta schemes for stiff systems of equations
+  [DOI: 10.1016/j.jcp.2019.05.014](https://doi.org/10.1016/j.jcp.2019.05.014)
+"""
+mutable struct PairedExplicitRK2 <: AbstractPairedExplicitRKSingle
+    const num_stages::Int
+
+    a_matrix::Matrix{Float64}
+    c::Vector{Float64}
+    b1::Float64
+    bS::Float64
+    c_end::Float64
+end # struct PairedExplicitRK2
+
+# Constructor that reads the coefficients from a file
+function PairedExplicitRK2(num_stages, base_path_monomial_coeffs::AbstractString,
+                           bS = 1.0, c_end = 0.5)
+    a_matrix, c = compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                            base_path_monomial_coeffs,
+                                                            bS, c_end)
+
+    b1 = 1.0 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a
+# semidiscretization
+function PairedExplicitRK2(num_stages, tspan, semi::AbstractSemidiscretization,
+                           verbose = false,
+                           bS = 1.0, c_end = 0.5)
```

Comment:
If there are sensible default values, I recommend to make them kwargs:
```suggestion
function PairedExplicitRK2(num_stages, tspan, semi::AbstractSemidiscretization;
                           verbose = false,
                           bS = 1.0, c_end = 0.5)
```